### ISSUE TITLE
FP S3966: Rule should consider the `leaveOpen` parameter

### ISSUE THREAD

**Issue by andrei-epure-sonarsource** [SonarSource/sonar-dotnet#2491]
### Description

Rule [Objects should not be disposed more than once](https://sonarcloud.io/organizations/budzow-github/rules?open=csharpsquid%3AS3966&q=S3966) raises an issue when `Dispose` would be called twice. However, we do not take into consideration the `leaveOpen` parameter of the `StreamWriter` class, which ensures that disposed will not be called on the `stream` parameter.

From the [StreamWriter documentation](https://docs.microsoft.com/en-us/dotnet/api/system.io.streamwriter.-ctor?view=netframework-4.8)

> leaveOpen : true to leave the stream open after the StreamWriter object is disposed; otherwise, false.

> Unless you set the leaveOpen parameter to true, the StreamWriter object calls Dispose() on the provided Stream object when StreamWriter.Dispose is called. 

### Repro steps

```
    public class ReproFromSonarCloud
    {
        public void S3966FalsePositive()
        {
            // Rule S3966 is a false positive:
            // The StreamWriter constructor used sets 'leaveOpen' to true, which prevents the
            // StreamWriter from disposing the MemoryStream. Dispose is only called due to the
            // 'using'-statement.

            using (MemoryStream memoryStream = new MemoryStream())
            using (StreamWriter writer = new StreamWriter(memoryStream, new System.Text.UTF8Encoding(false), 1024, leaveOpen: true)) // Noncompliant FP
            {
                //...
            }

        }
    }
```

### Expected behavior

The rule should check if the `leaveOpen` parameter is provided and `true` and should not raise this FP.

### Related information

* SonarC# Version: 7.15

This FP was raised by a customer who provided us with a very helpful [reproducer on SonarCloud](https://sonarcloud.io/code?id=ConsoleApp32&selected=ConsoleApp32%3AS3966.cs). Thank you!
